### PR TITLE
As a suggestion, rather spend a 404 page than 200.

### PR DESCRIPTION
**Pull request by tmloberon**
Logically, I'd find it if a controller or function wasn't found. A 404 page comes instead of a 200 Status. Because the route is wrong. Like a Wrong Url.

Triggered by:
- /index.php?cl=non-existent-controller
- /index.php?cl=start&fnc=non-existent-function

### CODE REVIEW

**Comment by Sieg** [comment]
Hi @tmloberon. Double checked how it works - If you have not existing controller situation, you will be redirected to the index page and exception will be logged, and only if your debug mode is on, you will see more information, and i dont see any issues about this workflow. Why you want to change it?

**Comment by tmloberon** [comment]
A non-existent controller is a non-existent page. We had a module that displayed its own page. We removed the module. The page no longer exists. Thus a 404 status. However Google still crawls over this link and it would be good if a status 404 comes instead of 200.

**Comment by keywan-ghadami-oxid** [comment]
my 2 cents on this
- this relates to my bug report https://bugs.oxid-esales.com/view.php?id=6370 . Using 404 makes the shop better from my point of view. the 404 page can offer a link to the start page or display products and more so the customer feels not lost. a simple redirect to startpage is just too confusing.
- handleRoutingException must be renamed or somehow changed if there is no exception used anymore

**Comment by tmloberon** [comment]
If this idea comes to approval. I like to program tests for it.

**Comment by tabsl** [comment]
+1 ... had a lot of exception spam like

`https://shop/category/product.html?cnid=a849897fd79efe4e86b24e9662a31742&listtype=list&fnc=showLogin`

--> redirect to startpage :-(

**Comment by jkrug** [comment]
+1

**Comment by flow-control** [comment]
+1

**Comment by flow-control** [comment]
Hey @tmloberon :vulcan_salute:

thank you very much for your pull request :+1: 
I talked with PO and we decided, that we want this feature, could you please fix travis and make it :green_heart:?

Thanks in advance

Flo

**Comment by TumTum** [comment]
Hi @flow-control, I am very happy that this idea was accepted. Have now finally written the tests. By corona I had forgotten that this PR was still open.

**Review by flow-control — CHANGES_REQUESTED** [review]

File: source/Core/ShopControl.php [hunk 1]
```diff
@@ -951,10 +950,29 @@ protected function getControllerClass($controllerKey)
         try {
             $controllerClass = $this->resolveControllerClass($controllerKey);
         } catch (\OxidEsales\Eshop\Core\Exception\RoutingException $exception) {
-            $this->handleRoutingException($exception);
-            $controllerClass = $controllerKey;
+            if ($this->isMetadataV1Controller($controllerKey)) {
+                $controllerClass = $controllerKey;
+            } else {
+                $this->handleRoutingException($exception);
+            }
```
[the remark this answers is not in the document]
Comment (tmloberon) [reply source/Core/ShopControl.php:957]: Yes sure, is also marked that the function should be deleted. As soon the controllers are no longer used by the Metadata v1. Now the question is, should the pull request for v6.2x also merged?

[the remark this answers is not in the document]
Comment (flow-control) [reply source/Core/ShopControl.php:957]: Currently it targets `master`, @Sieg and I can double check with the team. I'll ping you

**Comment by TumTum** [comment]
Hi @Sieg I have now updated the code and implemented what @flow-control  said.

**Comment by Sieg** [comment]
Hey guys, did some changes here, to handle the routing problems in one place, what do you think?

**Comment by Sieg** [comment]
Merged to master with some changes/improvements (https://github.com/OXID-eSales/oxideshop_ce/compare/85a2dd03e0136c21b1af04b63b5e0de5000b9184...71639371c6b21cd48ffe6169ace52ad3825cd9f4) Thanks.

**Comment by realFlowControl** [comment]
@Sieg WHOOP WHOOP :tada:

**Pull request closed without merging**